### DISCUSSION TITLE
Change password feature when using email authentication

### DISCUSSION THREAD

**Discussion by KomosSolutions** [serverpod/serverpod#2569] [Q&A] [answered]
Hi,

How would I create a 'change password' feature for a logged in user? 

I have implemented a password reset for a forgotten password, using the initatePasswordReset and resetPassword methods, but there doesn't seem to be a method on UserInfo on the serverside, or EmailAuthController on the client side to create a seamless way for a user to change their password after they've logged in.

**Comment by KomosSolutions** [comment, marked as answer]
For anyone who needs the same functionality I've hacked together a change password process using the reset password feature.  A built in feature needs to developed, but in the mean time you can : 

Create server endpoints:
- Create an 'initiate change password' endpoint which adds a fake verification code (123456) to the serverpod_email_reset table for the userID
- Create a 'cleanup' endpoint which removes the fake verification code from the serverpod_email_reset table

From the client:
- Call 'initiate change password' endpoint
- Call the serverpod built in method : emailAuthController.resetPassword(email, '123456', newPassword);
- Call the 'cleanup' endpoint

Hope this helps people while the lovely serverpod devs develop this feature.

**Comment by vlidholt** [comment]
We are planning some larger improvements to the authentication. This should definitely be a part of that!

**Reply by KellyDanielO** [reply]
the issue I have with this current flow is that there's no way to compare the old password
